extensions/sql-database-projects/src/models/project.ts

```diff
@@ -626,6 +626,29 @@ export class Project implements ISqlProject {
 		// make backup copy of project
 		await fs.copyFile(this._projectFilePath, this._projectFilePath + '_backup');
 
+		// remove Build includes and folder includes
+		const beforeFiles = this.files.filter(f => f.type === EntryType.File);
+		const beforeFolders = this.files.filter(f => f.type === EntryType.Folder);
+
+		// remove Build includes
+		for (const file of beforeFiles) {
+			// only remove build includes in the same folder as the project
+			if (!file.relativePath.includes('..')) {
+				await this.exclude(file);
```

Comment:
Since we're leveraging the existing functionality to exclude files from non-SDK project, what happens if this function fails halfway? Would user be left with a non-SDK project still but with some files excluded? If we can wrap the whole conversion in a transaction and "roll-back" if anything fails that'd be good

Reply:
Good call. I added rolling back if there was an error during the conversion and added a test.